synapse/storage/databases/main/end_to_end_keys.py

```diff
@@ -367,6 +367,57 @@ def _count_e2e_one_time_keys(txn):
             "count_e2e_one_time_keys", _count_e2e_one_time_keys
         )
 
+    async def set_e2e_fallback_keys(
+        self, user_id: str, device_id: str, fallback_keys: dict
```

Comment:
I believe that `fallback_keys` will be a `JsonDict` type. (from `synapse.types`, if it isn't imported already).